lib/plugins/aws/deploy/compile/events/apiGateway/lib/methods.js

```diff
@@ -23,13 +29,38 @@ module.exports = {
             const errorMessage = [
               `HTTP event of function ${functionName} is not an object nor a string.`,
               ' The correct syntax is: http: get users/list',
-              ' OR an object with "path" and "method" proeprties.',
+              ' OR an object with "path" and "method" properties.',
               ' Please check the docs for more info.',
             ].join('');
             throw new this.serverless.classes
               .Error(errorMessage);
           }
 
+          // Setup cors.
+          if (!!event.http.cors) {
+            corsEnabled = true;
+            if (typeof event.http.cors === 'object') {
+              // TODO - Some validation of cors object...
+              cors = event.http.cors;
+              cors.methods = cors.methods.map((m) => (m.toUpperCase()));
```

Comment:
Got an exception here:

TypeError: cors.methods.map is not a function
    at /app/lib/plugins/aws/deploy/compile/events/apiGateway/lib/methods.js:45:43
    at Array.forEach (native)
    at /app/lib/plugins/aws/deploy/compile/events/apiGateway/lib/methods.js:11:29
    at /app/node_modules/lodash/lodash.js:4411:15
    at baseForOwn (/app/node_modules/lodash/lodash.js:2654:24)
    at /app/node_modules/lodash/lodash.js:4380:18
    at Function.forEach (/app/node_modules/lodash/lodash.js:8634:14)
    at AwsCompileApigEvents.compileMethods (/app/lib/plugins/aws/deploy/compile/events/apiGateway/lib/methods.js:10:7)
    at AwsCompileApigEvents.tryCatcher (/app/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/app/node_modules/bluebird/js/release/promise.js:504:31)
    at Promise._settlePromise (/app/node_modules/bluebird/js/release/promise.js:561:18)
    at Promise._settlePromise0 (/app/node_modules/bluebird/js/release/promise.js:606:10)
    at Promise._settlePromises (/app/node_modules/bluebird/js/release/promise.js:685:18)
    at Async._drainQueue (/app/node_modules/bluebird/js/release/async.js:138:16)
    at Async._drainQueues (/app/node_modules/bluebird/js/release/async.js:148:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/app/node_modules/bluebird/js/release/async.js:17:14)